lib/composite_primary_keys/composite_predicates.rb

```diff
@@ -51,9 +51,43 @@ def cpk_join_predicate(table1, key1, table2, key2)
     end
 
     def cpk_in_predicate(table, primary_keys, ids)
-      and_predicates = ids.map do |id|
-        cpk_id_predicate(table, primary_keys, id)
+      if primary_keys.length == 2
+        low_cardinality_key_part, high_cardinality_key_part = if ids.map(&:second).uniq.size > ids.map(&:first).uniq.size
+                                                                %i[first second]
+                                                              else
+                                                                %i[second first]
+                                                              end
+
+        groups = ids.group_by(&low_cardinality_key_part).transform_values do |values|
```

Comment:
Since we are actually already creating hash tables keyed on id anyway (but with the 2nd column values instead), why not just do this at the start?

```
hash1 = Hash.new
hash2 = Hash.n32
ids.map.each do |value1, value2|
  hash1[value1] << value2
  hash2[value2] << value1
end

Then compare the number of keys in hash1 versus hash2 and dump the one with more keys. 

```